### ISSUE TITLE
ipcam stop issues

### ISSUE THREAD

**Issue by cuteyuyu** [AlexeyAB/darknet#922]
Excuse me, I use ipcam When the cam detects my object, can I stop to detect it?

**Comment by AlexeyAB** [comment]
Do you want to freeze window of Darknet-Yolo when any object is detected?

**Comment by cuteyuyu** [comment]
@AlexeyAB Yeah, is there a way to solve my idea?my classes=1,my ipcame only detects a target~~

**Comment by AlexeyAB** [comment]
If you use **yolov3.cfg or yolov3-tiny.cfg**, then 

* Add  `static int nboxes_demo = 0;` here: https://github.com/AlexeyAB/darknet/blob/ec766fc3f114dd4faaa3add500776a7bae957ba5/src/demo.c#L75
   

* Add     `nboxes_demo = nboxes;` here: https://github.com/AlexeyAB/darknet/blob/ec766fc3f114dd4faaa3add500776a7bae957ba5/src/demo.c#L121


* Replace this line: https://github.com/AlexeyAB/darknet/blob/ec766fc3f114dd4faaa3add500776a7bae957ba5/src/demo.c#L245
    to these two: 
```cpp
if(nboxes_demo > 0) cvWaitKey(0);
int c = cvWaitKey(1);
```

So when object will be detected, window will be frozen until you press any key.

**Comment by cuteyuyu** [comment]
@AlexeyAB  but I use yolov2-tiny.cfg for training , Is the method similar to yolov3-tiny.cfg??

**Comment by cuteyuyu** [comment]
@AlexeyAB  if I use  yolov2-tiny.cfg for training , is there a way to solve my idea?

**Comment by cuteyuyu** [comment]
@AlexeyAB I added an error message .The following figure shows the error
![d](https://user-images.githubusercontent.com/38469710/40758520-577960d8-64c0-11e8-8d26-65c685edd5d0.JPG)

May I ask which step was wrong?

**Comment by AlexeyAB** [comment]
Try to use this code:
```cpp
int i, j;
int cur_nboxes = 0;
	for (i = 0; i < nboxes; ++i) {
		for (j = 0; j < l.classes; ++j) {
			if (dets[i].prob[j] > demo_thresh) {
                           cur_nboxes++;
                        }
                  }
        }
nboxes_demo = cur_nboxes;
if(nboxes_demo > 0) {
  show_image_cv_ipl(det_img, "Demo");
  cvWaitKey(0);
}
```

**Comment by cuteyuyu** [comment]
@AlexeyAB  I have an idea!!My opinion is...
I use darknet_no_gpu and yolov2_tiny to train my weights
When my ipcam detected my object, ipcam stopped, and then used socket to send to Android (web client) to notify detected object . How can I make darknet a web server?

**Comment by AlexeyAB** [comment]
@cuteyuyu 
You can read these issues:
1. https://github.com/AlexeyAB/darknet/issues/571#issuecomment-378548435
2. https://github.com/AlexeyAB/darknet/issues/846#issuecomment-390399731
3. https://github.com/AlexeyAB/darknet/issues/605#issuecomment-379842681

**Comment by cuteyuyu** [comment]
@AlexeyAB  I haven't found the object yet. ipcam will stop automatically soon after . As shown below
![s](https://user-images.githubusercontent.com/38469710/40818464-ad042c76-6589-11e8-8d6e-b42bf47f56c5.JPG)
